crates/agglayer-storage/src/stores/state/mod.rs

```diff
@@ -381,6 +382,7 @@ impl StateStore {
         let mut nodes: Vec<Node<Keccak256Hasher>> = Vec::new();
         nodes.push(root_node);
 
+        let mut already_read = BTreeSet::new();
```

Comment:
Maybe we can find a better name, not a blocker tho

Reply:
I renamed it to `queued`, happy to rename it to something else

Reply:
I think that `queue` isn't the right name either, we're storing the `hash` of every node that the SMT contains, shouldn't it be something like `smt_hashes` or `node_hashes` ?
I also think that we could do the `if insert..` before even being in the match, but I'm fine letting it this way for now.